tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersHelper.cs

```diff
@@ -0,0 +1,512 @@
+            CodeownersEntry? existingEntry,
+            string path,
+            string serviceLabel,
+            List<string> serviceOwners,
+            List<string> sourceOwners,
+            bool isAdding);
+        string NormalizePath(string path);
+        CodeownersEntry findAlphabeticalInsertionPoint(List<CodeownersEntry> codeownersEntries, CodeownersEntry codeownersEntry);
+        public (List<CodeownersEntry>, int) mergeCodeownerEntries(List<CodeownersEntry> codeownersEntries, int index);
+        string addCodeownersEntryAtIndex(string codeownersContent, CodeownersEntry codeownersEntry, int index, bool codeownersEntryExists);
+        string formatCodeownersEntry(CodeownersEntry codeownersEntry);
+        (int StartLine, int EndLine) findBlock(string currentContent, string serviceCategory);
+        public string ExtractDirectoryName(string path);
+        string CreateBranchName(string prefix, string identifier);
+        
+        // Owner manipulation helper methods
+        List<string> AddUniqueOwners(List<string> existingOwners, List<string> ownersToAdd);
+        List<string> RemoveOwners(List<string> existingOwners, List<string> ownersToRemove);
+        List<string> ReplaceEntryInLines(string lines, CodeownersEntry targetEntry);
+    }
+
+    public class CodeownersHelper : ICodeownersHelper
+    {
+        public CodeownersEntry? FindMatchingEntries(IList<CodeownersEntry> entries, string path = null, string serviceLabel = null)
+        {
+            var codeownersEntries = new List<CodeownersEntry>();
+            foreach (var entry in entries)
+            {
+                // If serviceLabel is provided, match by ServiceLabels or PRLabels (exact match, case and space insensitive)
+                if (!string.IsNullOrEmpty(serviceLabel))
+                {
+                    // Check ServiceLabels for exact match
+                    if (entry.ServiceLabels?.Any(label => NormalizeLabel(label).Equals(NormalizeLabel(serviceLabel), StringComparison.OrdinalIgnoreCase)) == true)
+                    {
+                        return entry;
+                    }
+                    // Check PRLabels for exact match
+                    else if (entry.PRLabels?.Any(label => NormalizeLabel(label).Equals(NormalizeLabel(serviceLabel), StringComparison.OrdinalIgnoreCase)) == true)
+                    {
+                        return entry;
+                    }
+                }
+
+                // If repoPath is provided, match by PathExpression (contains match for paths - this is intentional)
+                if (!string.IsNullOrEmpty(path) && ExtractDirectoryName(entry.PathExpression).Contains(ExtractDirectoryName(path), StringComparison.OrdinalIgnoreCase) == true)
+                {
+                    return entry;
+                }
+            }
+            return null;
+        }
+
+        public (CodeownersEntry entry, bool existed) UpdateOrCreateCodeownersEntry(
```

Comment:
Nit: this action is typically named `CreateOrUpdate_____` instead of the other way.